tests/foreman/api/test_contentview.py

```diff
@@ -693,6 +700,92 @@ def test_positive_publish_with_puppet_multiple(self):
         for cvv in content_view.read().version:
             self.assertEqual(len(cvv.read().puppet_module), 1)
 
+    @tier2
+    def test_positive_publish_with_swid_tags(self):
+        """Verify SWID tags content file should exist in publish content view
+        version location
+
+        :id: 25f10e99-ceef-4543-9b9c-fc21690c088b
+
+        :steps:
+            1. create product and repository with custom contents having swid tags
+            2. sync the repository
+            3. create the content view
+            4. publish the content-view
+            5. ssh into Satellite
+            6. verify SWID tags content file exist in publish content view version location
+
+        :expectedresults: SWID tags content file should exist in publish content view
+            version location
+
+        :CaseAutomation: Automated
+
+        :CaseImportance: High
+
+        :CaseLevel: Integration
+        """
+        content_view = entities.ContentView(organization=self.org).create()
+        content_view.repository = [self.swid_repo]
+        content_view = content_view.update(['repository'])
+        content_view.publish()
+        content_view = content_view.read()
+        content_view_version_info = content_view.version[0].read()
+        self.assertEqual(len(content_view.repository), 1)
+        self.assertEqual(len(content_view.version), 1)
+        swid_repo_path = "{}/{}/content_views/{}/{}/custom/{}/{}/repodata".format(
+            CUSTOM_REPODATA_PATH,
+            self.org.name,
+            content_view.name,
+            content_view_version_info.version,
+            self.product.name,
+            self.swid_repo.name
+        )
+        result = ssh.command('ls {} | grep swidtags.xml.gz'.format(swid_repo_path))
```

Comment:
I don't think it's really a problem, but technically, this may pass even if file is not present:
```sh
$ touch 'fake swidtags.xml.gz'
$ ls |grep swidtags.xml.gz
fake swidtags.xml.gz
$ echo $?  # returns 0
```

If you want to verify that file exists in shell, use `test` (which is binary doing the same as `[` in conditionals):
```python
swid_repo_path = "{}/{}/content_views/{}/{}/custom/{}/{}/repodata/{}".format(
            CUSTOM_REPODATA_PATH,
            self.org.name,
            content_view.name,
            content_view_version_info.version,
            self.product.name,
            self.swid_repo.name,
            'swidtags.xml.gz',
        )
ssh.command('test -e {}'.format(swid_repo_path))
```

Reply:
The file name is like 85a1342224266f678aa8992c495baf1498829b7a9bbf1e6e7c022aa5d7524d15-swidtags.xml.gz, I don't want to add that long file name. Can I use a test to validate file without adding the exact file name?

Reply:
Oh, OK. I thought it's just `/swidtags.xml.gz`, without any checksum. You are right, adding this long and potentially unstable string to test is not good idea. And since we have to do partial match anyway, I think using `grep` on `ls` output is good approach.